### PR TITLE
core/store/migrate/migrations: add ccip cap spec migration

### PR DESCRIPTION
**Pull request by makramkd**
Migrations for the ccip capability spec. The actual Go part of this will be implemented in https://github.com/smartcontractkit/ccip because all the gethwrappers and other required components (e.g home chain reader) are there.

### CODE REVIEW

File: core/store/migrate/migrations/0243_ccip_capability_specs.sql [hunk 1]
```diff
@@ -0,0 +1,123 @@
+-- +goose Up
+-- +goose StatementBegin
+
+-- The ccip_specs table will hold the CCIP capability job specs.
+-- For each new CCIP capability version, we will create a new CCIP capability job.
+-- A single CCIP capability job manages all CCIP OCR instances across all chains.
+CREATE TABLE ccip_specs(
+    id BIGSERIAL PRIMARY KEY,
```
Comment (patrickhuie19) [inline core/store/migrate/migrations/0243_ccip_capability_specs.sql:8]: @ettec was looking at https://github.com/smartcontractkit/chainlink/commit/66d8d16638f16dc863cbc6d24356c3cf3c9724f0#diff-d9f073730e15e0db74de57a520f6d4ed076e3e72780cb1e423a42b8e8b6e3143 - should this use `BIGSERIAL` as well?

**Review by patrickhuie19 — APPROVED** [review]
lgtm

File: core/store/migrate/migrations/0243_ccip_capability_specs.sql (hunk 1 above)
Comment (makramkd) [reply core/store/migrate/migrations/0243_ccip_capability_specs.sql:8]: Oof actually the Go code uses `int32` which is just `SERIAL`. But I do see a lot of job specs with `BIGSERIAL` as the id and primary key 😬

Comment (makramkd) [reply core/store/migrate/migrations/0243_ccip_capability_specs.sql:8]: In practice I don't see us ever hitting this limit so will keep as-is. Any thoughts/objections?

**Review by dimkouv — APPROVED** [review]

**Pull request merged**